Sage/carbon
# refactor(pod): convert component code to typescript

### Proposed behaviour

- Refactor component, style and index files of `Pod` to typescript
- Reorganise `Pod` tests
    - Rewrite tests in terms of props as opposed to internal components to improve reliability
    - Replace all snapshot tests
- Refactor stories to typescript
- Supplementary changes
    - Ensure footer is not also aligned when `alignTitle` prop is set
    - Add aria labels to delete and undo buttons
    - Modify `with different sizes` story to include soft delete variants - so chromatic can check for changes.
    - Deprecate support for passing strings and objects to the `Pod`'s `onEdit` prop (see additional context for details)

### Current behaviour

- `Pod` is written in javascript
- `Pod` tests are written in terms of internal styled components

### Additional context

Currently, one accessibility violation with the `Pod` edit button: 

> Elements must only use allowed ARIA attributes

has been left intact. To address this the edit button must be rendered as a html `button` rather than an `a` link - but this first requires removed support for passing external links (as strings or objects) to the `onEdit` prop. Therefore, support for these two prop types has been deprecated in this PR with the intention of removal at a later date.

### Checklist




#### QA


### Testing instructions

- Verify no regressions were introduced from this refactor.
- Check all accessibility violations have been addressed, excluding one violation with edit button (see additional context).
- Verify that `Pod` footer no longer has same alignment as the title when the `alignTitle` prop is passed
